### ISSUE TITLE
Breakthrough damage is not shown in damage preview (but applied correctly) in BETA version (1.27)

### ISSUE THREAD

**Issue by zontik2012** [X2CommunityCore/X2WOTCCommunityHighlander#1289]
Steps to reproduce:
1) Get breakthrough tech like "conventional damage improvement"
2) Get in tactical
3) Aim at the enemy

Damage preview will show unmodified values, but hits themselves will do modified values (I was able to hit for 5 with conventional bullpup, but all previews showed 3-4)
I tried to remove Extended Information, thinking it might be the reason, but bug persisted with vanilla UI.
Other damage modifiers like ammo rounds or abilities work fine.
1.26.3 version does not have this bug

**Comment by Iridar** [comment]
Traced some code, apparently this check in `X2Effect_BonusWeaponDamage` fails:
```
DamageEffect = X2Effect_ApplyWeaponDamage(class'X2Effect'.static.GetX2Effect(AppliedData.EffectRef));
if( DamageEffect == none || DamageEffect.bIgnoreBaseDamage )
{
	return 0;
}
```
This happens because `X2Effect_ApplyWeaponDamage::GetDamagePreview()` that calls `X2Effect_BonusWeaponDamage::GetAttackingDamageModifier()` doesn't put anything into `AppliedData.EffectRef`.

The doesn't happen when the damage effect is actually applied, because `OnEffectAdded() -> CalculateDamageAmount()` which also calls `X2Effect_BonusWeaponDamage::GetAttackingDamageModifier()`, but passes its own `EffectAppliedData ApplyEffectParameters` to it. 

Since damage effect isn't actually getting applied during the ability damage preview step, there are no `ApplyEffectParameters` that can be passed to `GetAttackingDamageModifier()`.

This most likely is the reason behind `if( NewGameState != none )` check in the original `X2Effect_BonusWeaponDamage` class.

The super annoying part is all of that could be bypassed if `X2Effect_ApplyWeaponDamage` just passed itself to `GetAttackingDamageModifier()`.

**Comment by Iridar** [comment]
> The super annoying part is all of that could be bypassed if X2Effect_ApplyWeaponDamage just passed itself to GetAttackingDamageModifier().

Which unfortunately isn't an option, as this would break the subclasses.

**Comment by Iridar** [comment]
Well, it seems the problem is that until the effect is actually being applied to the target, the BonusWeaponDamage effect doesn't know which damage effect is looking it up. 

So the solutions are:
1. Revert this feature and let other mods deal with it.
2. Figure out how to fill the `EffectRef` struct in AppliedData so that BonusWeaponDamage can retrieve information it needs via existing means.
3. Add a separate CH_Improved version of GetAttackingDamageModifier and friends that will get the Damage Effect as one of the arguments.

**Comment by Iridar** [comment]
Gonna fix #612 properly later by adding a _CH version of GetAttackingDamage modifier, but it'll happen in 1.28. For now I'll just revert #1284, which causes this bug.
